packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -0,0 +1,1005 @@
+		commentId: Date.now(),
+		createdBy: currentUser,
+		comment,
+		createdAt: new Date().toISOString(),
+	} );
+
+	const confirmReplyComment = ( threadID ) => {
+		if ( threadID ) {
+			const newComment = generateReplyComment( commentReply );
+			apiFetch( {
+				path: '/wp/v2/comments/',
+				method: 'POST',
+				data: {
+					parent: threadID,
+					post: postId,
+					content: newComment.comment,
+					comment_date: newComment.createdAt,
+					comment_type: 'block_comment',
+					comment_author: currentUser,
+					comment_approved: 0,
+				},
+			} ).then( ( response ) => {
+				if ( 'trash' !== response.status && '' !== response.id ) {
+					setReplyMessage( true );
+				}
+			} );
+		}
+	};
+
+	return (
+		<>
+			{
+				// If there are no threads, show a message indicating no threads are available.
+				( ! Array.isArray( threads ) || threads.length === 0 ) && (
+					<VStack
+						className="editor-collab-sidebar__thread"
+						spacing="3"
+					>
+						<HStack
+							alignment="left"
+							spacing="3"
+							justify="flex-start"
+						>
+							{ __( 'No comments available' ) }
+						</HStack>
+					</VStack>
+				)
+			}
+
+			{ Array.isArray( threads ) &&
+				threads.length > 0 &&
+				threads.reverse().map( ( thread ) => (
+					<VStack
+						key={ thread.id }
+						className="editor-collab-sidebar__thread"
+						id={ thread.id }
+						spacing="2"
+					>
+						<HStack
+							alignment="left"
+							spacing="3"
+							justify="flex-start"
+						>
+							<img
+								src={ thread?.author_avatar_urls?.[ 48 ] }
+								className="editor-collab-sidebar__userIcon"
+								alt={ __( 'User avatar' ) }
+								width={ 32 }
+								height={ 32 }
+							/>
+							<VStack spacing="0">
+								<span className="editor-collab-sidebar__userName">
+									{ thread.author_name }
+								</span>
+								<time
+									dateTime={ format( 'h:i A', thread.date ) }
+									className="editor-collab-sidebar__usertime"
+								>
+									{ dateI18n( dateTimeFormat, thread.date ) }
+								</time>
+							</VStack>
+							<span className="editor-collab-sidebar__commentUpdate">
+								{ thread.status !== 'approved' && (
+									<HStack
+										alignment="right"
+										justify="flex-end"
+										spacing="0"
+									>
+										<Tooltip text={ __( 'Resolve' ) }>
+											<Button className="is-compact has-icon">
+												<Icon
+													icon={ published }
+													onClick={ () => {
+														setCommentConfirmation(
+															false
+														);
+														setShowConfirmation(
+															true
+														);
+														setShowConfirmationTabId(
+															thread.id
+														);
+													} }
+												/>
+											</Button>
+										</Tooltip>
+										{ 0 === thread.parent ? (
+											<DropdownMenu
+												icon={ moreVertical }
+												label="Select an action"
+												className="is-compact"
+												controls={ [
```

Comment:
We might want to cache the controls to avoid re-rendering the dropdown on every update.